web/modules/custom/itk_activity/src/Form/ActivityExternalSignupUserForm.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\itk_activity\Form\ActivityExternalSignupUserForm.
+ */
+
+namespace Drupal\itk_activity\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\itk_activity\Entity\ExternalUser;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Drupal\Core\Url;
+use Drupal\node\Entity\Node;
+
+/**
+ * ActivityExternalSignupUserForm.
+ */
+class ActivityExternalSignupUserForm extends FormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'itk_activity_external_signup_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, $nid = NULL) {
+    $node = \Drupal::entityTypeManager()->getStorage('node')->load($nid);
+
+    // If the node does not exist redirect to frontpage.
+    if (!isset($node)) {
+      drupal_set_message(t('The activity does not exist.'));
+      return new RedirectResponse(Url::fromRoute('<front>')->toString());
+    }
+
+    // Set node for later processing.
+    $form_state->set('node', $node);
+
+    $form['first_name'] = array(
+      '#type' => 'textfield',
+      '#required' => TRUE,
+      '#title' => $this->t('First name'),
+    );
+
+    $form['surname'] = array(
+      '#type' => 'textfield',
+      '#required' => TRUE,
+      '#title' => $this->t('Surname'),
+    );
+
+    $form['email'] = array(
+      '#type' => 'email',
+      '#required' => TRUE,
+      '#title' => $this->t('Email'),
+    );
+
+    $form['phone'] = array(
+      '#type' => 'tel',
+      '#required' => TRUE,
+      '#title' => $this->t('Phone'),
+    );
+
+    $form['submit'] = array(
+      '#type' => 'submit',
+      '#value' => $this->t('Sign up user'),
+    );
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $node = $form_state->get('node');
+
+    if (isset($node)) {
+      $firstName = $form_state->getValue('first_name');
+      $surname = $form_state->getValue('surname');
+      $email = $form_state->getValue('email');
+      $phone = $form_state->getValue('phone');
+
+      // Create new external user entity.
+      $externalUser = ExternalUser::create([
+        'type' => 'itk_activity_external_user',
+        'first_name' => $firstName,
+        'surname' => $surname,
+        'email' => $email,
+        'phone' => $phone,
+      ]);
+      $externalUser->save();
+
+      // Update activity node.
+      $node->field_external_signed_up_users[] = $externalUser;
+      $node->save();
+
+      // Add message.
+      drupal_set_message($firstName . t(' is registered to activity.'));
```

Comment:
This is not the currect usage of t(). Please see https://api.drupal.org/api/drupal/core%21includes%21bootstrap.inc/function/t/8.3.x

t($string, array $args = [], array $options = [])

:-)